spec/clj/tic_tac_toe_clojure/core_spec.clj

```diff
@@ -169,44 +241,60 @@
       (with-in-str "5" (pick-board-size "Fake prompt")))))
 
 (describe "A decision engine"
-  (it "gets the rows from a board"
+  (it "gets the rows from a 3x3 board"
     (should= [["X" "_" "_"] ["O" "_" "_"] ["X" "_" "_"]]
       (get-rows ["X" "_" "_"
                  "O" "_" "_"
-                 "X" "_" "_"])))
+                 "X" "_" "_"] 3)))
 
  (it "gets the rows from a 4x4 board"
    (should= [["X" "_" "_" "_"] ["O" "_" "_" "_"] ["X" "_" "_" "_"] ["O" "_" "_" "_"]]
-     (get-4x4-rows ["X" "_" "_" "_"
-                    "O" "_" "_" "_"
-                    "X" "_" "_" "_"
-                    "O" "_" "_" "_"])))
+     (get-rows ["X" "_" "_" "_"
+                "O" "_" "_" "_"
+                "X" "_" "_" "_"
+                "O" "_" "_" "_"] 4)))
 
   (it "gets the columns from a board"
     (should= [["X" "_" "_"] ["O" "_" "_"] ["X" "_" "_"]]
       (get-columns ["X" "O" "X"
                     "_" "_" "_"
-                    "_" "_" "_"])))
+                    "_" "_" "_"] 3)))
 
   (it "gets the columns from a 4x4 board"
     (should= [["X" "_" "_" "_"] ["O" "_" "_" "_"] ["X" "_" "_" "_"] ["O" "_" "_" "_"]]
-      (get-4x4-columns ["X" "O" "X" "O"
-                        "_" "_" "_" "_"
-                        "_" "_" "_" "_"
-                        "_" "_" "_" "_"])))
+      (get-columns ["X" "O" "X" "O"
+                    "_" "_" "_" "_"
+                    "_" "_" "_" "_"
+                    "_" "_" "_" "_"] 4)))
+
+  (it "should return a vector with the indices of the cells in a left to right diagonal for a 4x4 board"
+    (should= [0 5 10 15]
+      (get-left-diagonal 4)))
+
+  (it "should return a vector with the indices of the cells in a left to right diagonal for a 5x5 board"
+    (should= [0 6 12 18 24]
+      (get-left-diagonal 5)))
 
-  (it "gets the diagonals from a board"
+  (it "should return a vector with the indices of the cells in a right to left diagonal for a 4x4 board"
+    (should= [3 6 9 12]
+      (get-right-diagonal 4)))
+
+  (it "should return a vector with the indices of the cells in a right to left diagonal for a 5x5 board"
+    (should= [4 8 12 16 20]
+      (get-right-diagonal 5)))
+
+  (it "gets the diagonals from a 3x3 board"
       (should= [["X" "O" "X"] ["O" "O" "O"]]
         (get-diagonals ["X" "_" "O"
                         "_" "O" "_"
-                        "O" "_" "X"])))
+                        "O" "_" "X"] 3)))
 
   (it "gets the diagonals from a 4x4 board"
       (should= [["X" "X" "X" "X"] ["O" "O" "O" "O"]]
-        (get-4x4-diagonals ["X" "_" "_" "O"
-                            "_" "X" "O" "_"
-                            "_" "O" "X" "_"
-                            "O" "_" "_" "X"])))
+        (get-diagonals ["X" "_" "_" "O"
```

Comment:
Is there a way to make `get-diagonals` expecting only a board as an argument? can't this function find out the size of the board by itself?